Automattic/sensei
# Add usage tracking to emails

Resolves #6622.

## Proposed Changes
Logs the `sensei_email_send` event when an email is sent. This event has a `type` parameter to indicate the type of email that was sent.

Note that the format of the `type` parameter indicates both the recipient of the email as well as its type. For example, `learner-completed-course` indicates that the student was sent a course completed email. This naming convention was chosen so that it matches the names of the existing email types from #6228.

I tried to use the existing `IDENTIFIER_NAME` constant at first, but that didn't seem to be as straightforward as I was hoping, so I opted for a new constant instead.

I also tried to add a test case for ensuring the `sensei_log_event` function was called, but wasn't able to find a way.

## Testing Instructions
* Enable usage tracking.
* Enable all emails.
* Trigger each email notification.
* Check that a `sensei_email_send` event is logged in [Tracks](https://mc.a8c.com/tracks/live/?eventname=sensei_email_send&user=&useragent=). (You can filter by domain in the Username, ID, or Email field.)
* Make sure each event has the appropriate `type`.

## Pre-Merge Checklist
